streams/src/test/java/org/apache/kafka/streams/StreamsBuilderTest.java

```diff
@@ -1687,15 +1707,15 @@ public void shouldWrapProcessorsForStatelessOperators() {
             .peek((k, v) -> { }, Named.as("peek")) // wrapped 4
             .flatMapValues(e -> new ArrayList<>(), Named.as("flatMap")) // wrapped 5
             .toTable(Named.as("toTable")) // should be wrapped when we do StreamToTableNode
-            .filter((k, v) -> true, Named.as("filter-table")) // should be wrapped once we do TableProcessorNode
```

Comment:
to be honest I had no idea this was a stateful operation when I wrote this test, as you can tell from the name being `shouldWrapProcessorsForStatelessOperators` -- we should either change the name or cut out the `.toTable().filter().toStream()` part

Honestly either is fine with me, since we have a separate test for the KTable filter now. Though I kind of like the idea of having one test with multiple operators for the KTable stuff, since you never know how things might interact in the DSL. Maybe just move this part to a separate test and we can have one test that covers all the stateful KTable operators you're doing here (eg mapValues, filter, etc)